### PR TITLE
Be more specific in findobj return type

### PR DESCRIPTION
**Pull request by ksunden**
## PR summary

This just separates the case where we know more about the the return type of `findobj`.

This was requested by https://github.com/pydata/xarray/pull/8889#discussion_r1544463148

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]

**Comment by rcomer** [comment]
This one has two approvals.  Was it just waiting for CI?

**Pull request merged**
